daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -153,32 +152,14 @@ final case class Compiler(packages: PackageId PartialFunction Package) {
     * The packages do not need to be in any specific order, as long as they and all the packages
     * they transitively reference are in the [[packages]] in the compiler.
     */
-  @throws(classOf[PackageNotFound])
-  def compilePackages(toCompile0: Iterable[PackageId]): Map[SDefinitionRef, SExpr] = {
-    var defns = Map.empty[SDefinitionRef, SExpr]
-    val compiled = mutable.Set.empty[PackageId]
-    var toCompile = toCompile0.toList
-    val foundDependencies = mutable.Set.empty[PackageId]
-
-    while (toCompile.nonEmpty) {
-      val pkgId = toCompile.head
-      toCompile = toCompile.tail
-      try {
-        if (!compiled.contains(pkgId))
-          defns ++= compilePackage(pkgId)
-      } catch {
-        case PackageNotFound(dependency) if dependency != pkgId =>
-          if (foundDependencies.contains(dependency)) {
-            throw CompileError(s"Cyclical packages, stumbled upon $dependency twice")
-          }
-          foundDependencies += dependency
-          toCompile = dependency :: pkgId :: toCompile
-      }
-      compiled += pkgId
-    }
-
-    defns
-  }
+  @throws[PackageNotFound]
+  @throws[CompileError]
+  @throws[ValidationError]
+  def compilePackages(toCompile: Iterable[PackageId]): Map[SDefinitionRef, SExpr] =
+    dependenciesInTopologicalOrder(toCompile.toList, packages)
```

Comment:
Might we worth explaining that we use `dependenciesInTopologicalOrder` because packages need to be compiled in order. Perhaps that's obvious to others though.

Reply:
I add a sentence. 
This is very common anytime you need to handle dependencies. Thinks for example to Maekfile.

Reply:
done